core/store/src/config.rs

```diff
@@ -86,24 +85,118 @@ impl StoreConfig {
         }
     }
 
-    pub const fn read_only() -> StoreConfig {
-        StoreConfig::const_default().with_read_only(true)
+    /// Returns cache size for given column.
+    pub const fn col_cache_size(&self, col: crate::DBCol) -> bytesize::ByteSize {
+        match col {
+            crate::DBCol::State => self.col_state_cache_size,
+            _ => bytesize::ByteSize::mib(32),
+        }
     }
+}
 
-    pub const fn read_write() -> StoreConfig {
+impl Default for StoreConfig {
+    fn default() -> Self {
         Self::const_default()
     }
+}
+
+// TODO(#6857): Make this method in StoreOpener.  This way caller won’t need to
+// resolve path to the storage.
+pub fn store_path_exists<P: AsRef<std::path::Path>>(path: P) -> bool {
+    std::fs::canonicalize(path).is_ok()
+}
+
+// TODO(#6857): Get rid of this function.  Clients of this method should not
+// care about path of the storage instead letting StoreOpener figure that one
+// out.
+pub fn get_store_path(base_path: &std::path::Path) -> std::path::PathBuf {
+    base_path.join(STORE_PATH)
+}
+
+/// Builder for opening a RocksDB database.
+///
+/// Typical usage:
+///
+/// ```ignore
+/// let store = StoreOpener::new(&near_config.config.store)
+///     .home(neard_home_dir)
+///     .open();
+/// ```
+pub struct StoreOpener<'a> {
+    path: Option<&'a std::path::Path>,
```

Comment:
The purpose of the `path` field is unclear. Further below you have a TODO to remove it. Anyway, please mention that `path` has a priority over `home`.